docs/core/extensions/httpclient-factory-troubleshooting.md

```diff
@@ -0,0 +1,184 @@
+
+- Recycling and recreating `HttpMessageHandler`'s when their lifetime expires is essential for `IHttpClientFactory` to ensure the handlers react to DNS changes. `HttpClient` is tied to a specific handler instance upon its creation, so new `HttpClient` instances should be requested in a timely manner to ensure the client will get the updated handler.
+
+- Disposing of such `HttpClient` instances **created by the factory** will not lead to socket exhaustion, as its disposal **will not** trigger disposal of the `HttpMessageHandler`. `IHttpClientFactory` tracks and disposes of resources used to create `HttpClient` instances, specifically the `HttpMessageHandler` instances, as soon their lifetime expires and there's no `HttpClient` using them anymore.
+
+_Typed clients_ are intended to be **short-lived** as well, as an `HttpClient` instance is injected into the constructor, so it will share the _typed client_ lifetime.
+
+For more information, see the [`HttpClient` lifetime management](httpclient-factory.md#httpclient-lifetime-management) and [Avoid _typed clients_ in singleton services](httpclient-factory.md#avoid-typed-clients-in-singleton-services) sections in the `IHttpClientFactory` guidelines.
+
+## `HttpClient` uses too many sockets
+
+Even if `IHttpClientFactory` is used, it is still possible to hit the socket exhaustion issue with a specific usage scenario. By default, `HttpClient` doesn't limit the amount of concurrent requests. If a large amount of HTTP/1.1 requests are started concurrently at the same time, each of them will end up triggering a new HTTP connection attempt, because there is no free connection in the pool and no limit is set.
+
+❌ DO NOT start a large amount of HTTP/1.1 requests concurrently at the same time without specifying the limits.
+
+✔️ CONSIDER setting <xref:System.Net.Http.HttpClientHandler.MaxConnectionsPerServer?displayProperty=nameWithType> (or <xref:System.Net.Http.SocketsHttpHandler.MaxConnectionsPerServer?displayProperty=nameWithType>, if you use it as a primary handler) to a reasonable value. Note that these limits only apply to the specific handler instance.
+
+✔️ CONSIDER using HTTP/2 which allows multiplexing requests over a single TCP conneciton.
+
+## _Typed client_ gets a wrong `HttpClient` injected
+
+There can be various situations in which it is possible to get an unexpected `HttpClient` injected into a _typed client_. Most of the time, the root cause will be in an erroneous configuration, as, by DI design, any subsequent registration of a service overrides the previous one.
+
+_Typed clients_ use _named clients_ "under the hood": adding a _typed client_ implicitly registers and links it to a _named client_. The client name, unless explicitly provided, will be set to the type name of `TClient`. Note that this would be the **first** one from the `TClient,TImplementation` pair in case `AddHttpClient<TClient,TImplementation>` overloads are used.
```

Comment:
```suggestion
_Typed clients_ use _named clients_ "under the hood": adding a _typed client_ implicitly registers and links it to a _named client_. The client name, unless explicitly provided, will be set to the type name of `TClient`. This would be the **first** one from the `TClient,TImplementation` pair if `AddHttpClient<TClient,TImplementation>` overloads are used.
```